packages/dds/sequence/src/sharedString.ts

```diff
@@ -223,6 +224,7 @@ export class SharedString
 	 * @param combiningOp - Optional. Specifies how to combine values for the property, such as "incr" for increment.
 	 */
 	public annotateMarker(marker: Marker, props: PropertySet, combiningOp?: ICombiningOp) {
+		assert(props.markerId === undefined, "Cannot change the markerId of an existing marker");
```

Comment:
Why is this check in sharedString? It seems like it should be in mergeTree where the limitation is. Also, not all code paths that could set the markerId are protected here.